chia/rpc/wallet_rpc_api.py

```diff
@@ -1905,21 +1938,21 @@ async def create_signed_transaction(self, request, hold_lock=True) -> EndpointRe
                     puzzle_announcements_to_consume=puzzle_announcements,
                     min_coin_amount=min_coin_amount,
                 )
-        else:
-            signed_tx = await self.service.wallet_state_manager.main_wallet.generate_signed_transaction(
-                amount_0,
-                bytes32(puzzle_hash_0),
-                fee,
-                coins=coins,
-                exclude_coins=exclude_coins,
-                ignore_max_send_amount=True,
-                primaries=additional_outputs,
-                memos=memos_0,
-                coin_announcements_to_consume=coin_announcements,
-                puzzle_announcements_to_consume=puzzle_announcements,
-                min_coin_amount=min_coin_amount,
-            )
-        return {"signed_tx": signed_tx.to_json_dict_convenience(self.service.config)}
+                return {"signed_tx": tx.to_json_dict_convenience(self.service.config)}
```

Comment:
We should return `{"signed_tx": blah, "signed_txs": [blah]}` here for backwards compatibility but also reducing the complexity on the client moving forward as most wallets will return multiple transactions.

Reply:
Added backwards compatibility

Reply:
I meant the other way around.  The CAT side can only return signed_txs, we should just make the XCH case return signed_txs as well.  That was you don't need the Union and if/else in the RPC client.

Reply:
Oh I see. That certainly makes sense.

Reply:
Marked as resolved.